compiler/src/dotty/tools/dotc/transform/Memoize.scala

```diff
@@ -96,10 +96,12 @@ class Memoize extends MiniPhase with IdentityDenotTransformer { thisPhase =>
         case _ => ()
       }
 
-    def removeAnnotations(denot: SymDenotation): Unit =
+    def removeUnwantedAnnotations(denot: SymDenotation): Unit =
       if (sym.annotations.nonEmpty) {
         val cpy = sym.copySymDenotation()
-        cpy.annotations = Nil
+        // Keep @deprecated annotation so that accessors can
+        // be marked as deprecated in the bytecode
```

Comment:
It's probably worth leaving a TODO here that instead of hardcoding which annotations to keep, we should respect the meta-annotations: https://www.scala-lang.org/api/2.12.3/scala/annotation/meta/index.html, for example `@deprecated` is defined with a bunch of them: https://github.com/scala/scala/blob/be050b224d5a7e156ee6916dd29dd3952df25e36/src/library/scala/deprecated.scala#L59

Reply:
TODOed :upside_down_face: